spec/lang/step.md

```diff
@@ -558,13 +558,13 @@ impl<M: Memory> Machine<M> {
         // Evaluate all arguments.
         let arguments = arguments.try_map(|arg| self.eval_value(arg))?;
 
-        let (value, ret_type) = self.eval_intrinsic(intrinsic, arguments)?;
+        let (value, _ret_ty) = self.eval_intrinsic(intrinsic, arguments)?;
 
         if let Some(ret_place) = ret_place {
-            let align = Align::max_for_offset(ret_type.size::<M>()).unwrap();
-            
-            let bytes = Type::encode::<M>(ret_type, value);
-            self.mem.store(ret_place, bytes, align)?;
+            let ret_expr = ret_expr.unwrap();
```

Comment:
Oh now I see what you mean... we should really just make `eval_place` return the type. But that's for a future change.

Reply:
Wouldn't this introduce a lot of redundancy?

Reply:
Yes. But that might be the lesser evil.